ot/gromov/_semirelaxed.py

```diff
@@ -1212,42 +1524,131 @@ def semirelaxed_fgw_barycenters(
+        lambdas = nx.ones(S) / S
+    else:
+        lambdas = list_to_array(lambdas)
+        lambdas = nx.from_numpy(lambdas)
 
     if fixed_structure:
         if init_C is None:
             raise UndefinedParameter(
                 'If C is fixed it must be provided in init_C')
         else:
             C = init_C
-    else:
-        if init_C is None:
-            rng = check_random_state(random_state)
-            xalea = rng.randn(N, 2)
-            C = dist(xalea, xalea)
-            C = nx.from_numpy(C, type_as=ps[0])
-        else:
-            C = init_C
 
     if fixed_features:
         if init_X is None:
             raise UndefinedParameter(
                 'If X is fixed it must be provided in init_X')
         else:
             X = init_X
+
+    # Initialization of transport plans, C and X (if not provided by user)
+    if G0 in ['product', 'random_product', 'random']:
+        # both init_X and init_C are simply deduced from transport plans
+        # if not initialized
+        if init_C is None:
+            init_C = nx.zeros((N, N), type_as=Cs[0])  # to know the barycenter shape
+
+            T = [semirelaxed_init_plan(
+                Cs[i], init_C, ps[i], method=G0, use_target=False,
+                random_state=random_state, nx=nx) for i in range(S)]
+
+            C = update_barycenter_structure(
+                T, Cs, lambdas, loss_fun=loss_fun, nx=nx)
+            if G0 in ['product', 'random_product']:
+                # initial structure is constant so we add a small random noise
+                # to avoid getting stuck at init
+                np.random.seed(random_state)
+                noise = np.random.uniform(-0.01, 0.01, size=(N, N))
+                if symmetric:
+                    noise = (noise + noise.T) / 2.
+                noise = nx.from_numpy(noise)
+                C = C + noise
+
+        else:
+            T = [semirelaxed_init_plan(
+                Cs[i], init_C, ps[i], method=G0, use_target=False,
+                random_state=random_state, nx=nx) for i in range(S)]
+
+            C = init_C
+
+        if init_X is None:
+            X = update_barycenter_feature(
+                T, Ys, lambdas, loss_fun=loss_fun, nx=nx)
+        else:
+            X = init_X
+
+        Ms = [dist(Ys[s], X) for s in range(len(Ys))]
+
     else:
+        # more computationally costly inits could be used on structures
+        # so we assume affordable a Kmeans-like init for features
+        # and use it by default.
+
         if init_X is None:
-            X = nx.zeros((N, d), type_as=ps[0])
+            stacked_features = nx.concatenate(Ys, axis=0)
+            if sklearn_import:
+                stacked_features = nx.to_numpy(stacked_features)
+                km = KMeans(n_clusters=N, random_state=random_state,
+                            n_init=1).fit(stacked_features)
+                init_X = nx.from_numpy(km.cluster_centers_)
+            else:
+                warnings.warn(
```

Comment:
raise error: install sklearn for kmeans (redo a better texte) and tests that teh error is raised